### PR TITLE
refactor: simplify connect Resolve... funcs

### LINKED ISSUE #385: [refactor] Reduce duplication in connect_service.go
### Requirements

Recently, a [line was removed](https://github.com/open-feature/flagd/pull/375) by accident in a refactoring. This was easily missed as there is a lot of code duplication for each `ResolveBoolean`,`ResolveString`, ... method.
Simplify these functions by reducing duplication.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

- Reduces duplicated code in Resolve... funcs

### Related Issues

Fixes #385 

### How to test

Tested against integration tests.

### CODE REVIEW

File: pkg/service/connect_service.go [hunk 1]
```diff
@@ -245,170 +245,103 @@ func (s *ConnectService) Notify(n Notification) {
...
-	ctx context.Context,
-	req *connect.Request[schemaV1.ResolveBooleanRequest],
-) (*connect.Response[schemaV1.ResolveBooleanResponse], error) {
+type response[T constraints] interface {
+	SetResult(value T, variant, reason string) error
+}
+
+type constraints interface {
+	bool | string | map[string]any | float64 | int64
+}
```
Comment (Kavindu-Dodan) [inline pkg/service/connect_service.go:254]: Not a must but in my view these interfaces fits best inside `connect_service_types.go` :)

**Comment by Kavindu-Dodan** [comment]
Overall looks good 👍

File: pkg/service/connect_service.go (hunk 1 above)
Comment (skyerus) [reply pkg/service/connect_service.go]: Good suggestion, done :)

**Review by james-milligan — APPROVED** [review]
LGTM

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**